src/Sarif/Baseline/V2/V2ResultMatcher.cs

```diff
@@ -115,6 +115,43 @@ private void LinkResultsWithIdenticalWhere()
             }
         }
 
+        private void LinkFirstAndLastFromSameArtifact()
+        {
+            int afterIndex = 0;
+            int beforeIndex = 0;
+
+            while (beforeIndex < Before.Count && afterIndex < After.Count)
+            {
+                // Get the next After Result (the first for a given Uri)
+                ExtractedResult afterFirstForUri = After[afterIndex];
+
+                // Look for the first Before Result with the same Uri, if any
+                ExtractedResult beforeFirstForUri = FirstWithUri(afterFirstForUri, Before, ref beforeIndex);
+
+                // If there was one...
+                if (beforeFirstForUri != null)
+                {
+                    // ... Try to link the first Results together
```

Comment:
>together [](start = 57, length = 8)

Do we have to worry that the identified pair of before/after results might already be linked because the first phase found that they had identical "where"? Or even if we don't have to "worry" about it, could we improve perf by noticing that they are already linked and avoiding the calls to `LinkIfSimilar`?

Reply:
LinkIfSimilar ensures the results are not already linked. Better to have it there to ensure it happens no matter the caller. I suspect the runtime inlines the whole function, since it's quite short, so I don't think there will be any performance impact.